modules/git/git.go

```diff
@@ -161,68 +180,82 @@ func Init(ctx context.Context) error {
-		if err := checkAndSetConfig(configKey, defaultValue, false); err != nil {
+	// Git requires setting user.name and user.email in order to commit changes - old comment: "if they're not set just add some defaults"
+	// TODO: need to confirm whether users really need to change these values manually. It seems that these values are dummy only and not really used.
+	// If these values are not really used, then they can be set (overwritten) directly without considering about existence.
+	for configKey, defaultValue := range map[string]string{
+		"user.name":  "Gitea",
+		"user.email": "gitea@fake.local",
+	} {
+		if err := configSetNonExist(configKey, defaultValue); err != nil {
 			return err
 		}
 	}
 
 	// Set git some configurations - these must be set to these values for gitea to work correctly
-	if err := checkAndSetConfig("core.quotePath", "false", true); err != nil {
+	if err := configSet("core.quotePath", "false"); err != nil {
 		return err
 	}
 
 	if CheckGitVersionAtLeast("2.10") == nil {
-		if err := checkAndSetConfig("receive.advertisePushOptions", "true", true); err != nil {
+		if err := configSet("receive.advertisePushOptions", "true"); err != nil {
 			return err
 		}
 	}
 
 	if CheckGitVersionAtLeast("2.18") == nil {
-		if err := checkAndSetConfig("core.commitGraph", "true", true); err != nil {
+		if err := configSet("core.commitGraph", "true"); err != nil {
 			return err
 		}
-		if err := checkAndSetConfig("gc.writeCommitGraph", "true", true); err != nil {
+		if err := configSet("gc.writeCommitGraph", "true"); err != nil {
 			return err
 		}
 	}
 
 	if CheckGitVersionAtLeast("2.29") == nil {
 		// set support for AGit flow
-		if err := checkAndAddConfig("receive.procReceiveRefs", "refs/for"); err != nil {
+		if err := configAddNonExist("receive.procReceiveRefs", "refs/for"); err != nil {
 			return err
 		}
 		SupportProcReceive = true
 	} else {
-		if err := checkAndRemoveConfig("receive.procReceiveRefs", "refs/for"); err != nil {
+		if err := configUnsetAll("receive.procReceiveRefs", "refs/for"); err != nil {
 			return err
 		}
 		SupportProcReceive = false
 	}
 
-	if runtime.GOOS == "windows" {
-		if err := checkAndSetConfig("core.longpaths", "true", true); err != nil {
+	if CheckGitVersionAtLeast("2.35.2") == nil {
+		// since Git 2.35.2, git adds a protection for CVE-2022-24765, the protection denies the git directories which are not owned by current user
+		// however, some docker users and samba users (maybe more, issue #19455) have difficulty to set their Gitea git repositories to the correct owner.
+		// the reason behind the problem is: docker/samba uses some uid-mapping mechanism, which are unstable/unfixable in some cases.
+		// now Gitea always use its customized git config file, and all the accesses to the git repositories can be managed,
+		// so it's safe to set "safe.directory=*" for internal usage only.
+		if err := configSet("safe.directory", "*"); err != nil {
```

Comment:
I think this should be made an option, e.g. `git.UNSAFE_DIRECTORY = true` and default it to `false`. It's a workaround, not something that should be done unconditionally as you never know whether gitea runs in single-user or multi-user context (where the security issue may still apply).

Reply:
I can not see a security issue here, `and all the accesses to the git repositories can be managed`, so I think it's not worth to introduce so many options to make the config file more and more complex.